src/main/java/com/marmelab/quoridor/model/Board.java

```diff
@@ -52,26 +53,39 @@ public int getBoardSize() {
         return squares;
     }
 
-    public void addFence(final Fence fence) {
+    public void addFence(final Fence fence, final Pawn pawn) {
         final FenceBlock fenceBlock = new FenceBlock(fence.getNorthwestTile());
         final PositionTile positionTile = new PositionTile(fence.getNorthwestTile());
         if (hasAlreadyAFenceAtTheSamePosition(fence.getNorthwestTile()) || hasNeighbourFence(fence.isHorizontal(), positionTile)) {
             LOGGER.info("Intersection: {}", fence);
         } else {
-            addFence(fence, fenceBlock);
+            addFence(fence, fenceBlock, pawn);
         }
     }
 
-    private void addFence(final Fence fence, final FenceBlock fenceBlock) {
-        fences.add(new Fence(fence));
+    private void addFence(final Fence fence, final FenceBlock fenceBlock, final Pawn pawn) {
```

Comment:
I think having 3 methods named the same is quite confusing here, they're not "proxies" of each other (as in with different parameters) but do different things, right?